python/lsst/meas/mosaic/mosaicfit/mosaicfit.cc

```diff
@@ -0,0 +1,290 @@
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the LSST License Statement and
+ * the GNU General Public License along with this program.  If not,
+ * see <https:
+ */
+#include "pybind11/pybind11.h"
+#include "pybind11/stl.h"
+
+#include "numpy/arrayobject.h"
+#include "ndarray/pybind11.h"
+
+#include "lsst/meas/mosaic/mosaicfit.h"
+
+namespace py = pybind11;
+using namespace pybind11::literals;
+
+namespace lsst {
+namespace meas {
+namespace mosaic {
+
+namespace {
+
+void declareSource(py::module &mod) {
+    using Class = Source;
+    using PyClass = py::class_<Class, std::shared_ptr<Class>>;
+
+    PyClass cls(mod, "Source");
+
+    cls.attr("UNSET") = py::cast(static_cast<int>(Class::UNSET));
+
+    cls.def(py::init<lsst::afw::table::SourceRecord const &>(), "record"_a);
+    cls.def(py::init<lsst::afw::table::SimpleRecord const &, lsst::afw::image::Wcs const &>());
+    cls.def(py::init<lsst::afw::coord::Coord, double>(), "coord"_a,
+            "flux"_a = std::numeric_limits<double>::quiet_NaN());
+    cls.def(py::init<typename Source::IdType, typename Source::ChipType, typename Source::ExpType, double,
+                     double, double, double, double, double, double, double, bool>(),
+            "id"_a, "chip"_a, "exp"_a, "ra"_a, "dec"_a, "x"_a, "xerr"_a, "y"_a, "yerr"_a, "flux"_a,
+            "fluxerr"_a, "astromBad"_a);
+
+    cls.def("getId", &Class::getId);
+    cls.def("getChip", &Class::getChip);
+    cls.def("getExp", &Class::getExp);
+    cls.def("getSky", &Class::getSky);
+    cls.def("getRa", &Class::getRa);
+    cls.def("getDec", &Class::getDec);
+    cls.def("getPixels", &Class::getPixels);
+    cls.def("getX", &Class::getX);
+    cls.def("getY", &Class::getY);
+    cls.def("getXErr", &Class::getXErr);
+    cls.def("getYErr", &Class::getYErr);
+    cls.def("getFlux", &Class::getFlux);
+    cls.def("getFluxErr", &Class::getFluxErr);
+    cls.def("getAstromBad", &Class::getAstromBad);
+    cls.def("setChip", &Class::setChip);
+    cls.def("setExp", &Class::setExp);
+    cls.def("setFlux", &Class::setFlux);
+}
+
+void declarePoly(py::module &mod) {
+    using Class = Poly;
+    using PyClass = py::class_<Class, std::shared_ptr<Class>>;
+
+    PyClass cls(mod, "Poly");
+
+    cls.def_readwrite("order", &Class::order);
+    cls.def_readwrite("ncoeff", &Class::ncoeff);
+    cls.def_readwrite("xorder", &Class::xorder, py::return_value_policy::copy);
+    cls.def_readwrite("yorder", &Class::yorder, py::return_value_policy::copy);
```

Comment:
As with the `xorder` and `yorder` attributes on `FluxFitParams`, these should not be wrapped, as they'll only cause trouble in Python.